### ISSUE TITLE
visualize by clusters

### ISSUE THREAD

**Issue by bernstei** [lab-cosmo/chemiscope#370]
Is there any way to select more than one atom at a time? I'd like to figure out what's in common to the atoms in a particular cluster that I see, and I hoped chemiscope would facilitate that, but other than showing one atom at a time, I don't see any way to map back from the visualized _clustering_ in the PCA (which I'm using a my axes) to the atomic structures.

**Comment by ceriottm** [comment]
There are many ways you could do this. One is opening multiple structure frames so you can compare many structures (click on the "duplicate viewer" icon on the structure panel) ![Image](https://github.com/user-attachments/assets/6763259b-110d-4a02-a484-0199ab266a71)

another is that - if you define the clustering at the environment level - you can *color* the atoms by property, see e.g. 
https://chemiscope.org/docs/examples/4-colors.html#sphx-glr-examples-4-colors-py

**Comment by ceriottm** [comment]
Do either of these approaches solve your problem or I misunderstood what you were looking for? In case they do, any suggestion on how to make these options more findable?

**Comment by bernstei** [comment]
I want to highlight (in the geometry viewer) the 100 that are in one cluster, to distinguish them from the few hundred in the other cluster.  The duplicate viewer approach seems like it'd allow me to view maybe 2 or 3 atomic environments at a time, but that doesn't seem like enough.

The clustering at the input file level _is_ an option, but it's not perfect.  I'd have to either run a generic clustering algorithm on the reduced coordinates, or perhaps look at the initial visualization and figure out a range of coordinates (box, sphere?), and go back and manually select those to set a new property in the input file, then visualize again and see if the clustering I did corresponds to the clusters I see by eye in the visualizer.  That's useful, certainly, but it'd be nice to be able to somehow output all the atoms that correspond to visualized points that are enclosed in a dragged out rectangle.  Ideally, I'd like to see them in the chemiscope structure viewer (admittedly you'd have to do something about the fact that they could come from different structures - maybe allow the user to select which structure is being viewed, and show it with the selected atoms highlighted?), or maybe export their indices (so I could go back to my usual geometry viewer and highlight them there).

**Comment by Luthaf** [comment]
We have a couple of issues about sub-selecting points on the map: #3 (which could also be expanded to a manual "lasso" selection), #4, and #53. 

Unfortunately, all of them would require a significant amount of work and bookkeeping in chemiscope, which is why nothing has been implemented yet. There was a WIP branch doing some of the rendering work for the map in #113, but did not include the user interface for selecting which points should be "hidden". There is no existing code to do this in the structure viewer, so it would have to be written from scratch.

**Comment by ceriottm** [comment]
Have you tried using chemiscope as a jupyter widget? I think the workflow you discuss here would be much more easily realized having the possibility of just refreshing the viewer quickly without having to go modify->write_input->load.
It might also be possible to use (and extend) the stateful widget to make this more interactive without having to implement a selection UI, that would go far beyond what is possible at present.

**Comment by bernstei** [comment]
I agree, it would definitely be better in a more interactive setting like jupyter, so I'll try that. But it still leaves it as solving an inverse problem - guess a possible clustering, see if it corresponds to what you see on the map, instead of defining the cluster in the visualized map and saving that information somehow.

**Comment by rosecers** [comment]
This might be partially addressed in #499

**Comment by ceriottm** [comment]
I think that with atom coloring by property, the ability to highlight a subset of the points on the map, and the possibility of having multiple viewers open, we can consider this solved.
